dali/base/dadfs.cpp

```diff
@@ -9204,7 +9205,7 @@ GetFileClusterNamesType CDistributedFileDirectory::getFileClusterNames(const cha
 // --------------------------------------------------------
 
 
-static CDistributedFileDirectory *DFdir = NULL;
+static std::atomic<CDistributedFileDirectory *> DFdir{nullptr};
```

Comment:
Similar comments to above - either make it atomic, or protect all access within a critical section.